src/xbot/common/command/BaseRobot.java

```diff
@@ -132,6 +154,17 @@ public void testPeriodic() {
     protected void sharedPeriodic() {
         xScheduler.run();
         this.updatePeriodicDataSources();
+        
+        loopCycleCounter++;
+        double timeSinceLastLog = Timer.getFPGATimestamp() - lastFreqCounterResetTime;
+        if(lastFreqCounterResetTime > 0 && timeSinceLastLog >= frequencyReportInterval.get()) {
```

Comment:
I think `lastFreqCounterResetTime` starts as -1 and is only set to something larger inside this `if` so it would never actually execute?